### ISSUE TITLE
Requests 'table' UI hangs while testing many requests

### ISSUE THREAD

**Issue by alias1** [SecurityInnovation/AuthMatrix#23]
The UI doesn't update while testing, and there is basically no indication of whether it's working or has crashed until it is all finished and updates.

It would be good to have some kind of visual indication, either updating each request line/role as it processes, or a progress bar showing how far through it is (possibly with the option to pause/abort)

**Comment by mickayz** [comment]
While it is less than ideal, the current version 0.6.2 has two UI indications for when requests are running.  

The first is that the buttons on the bottom row will all be disabled aside from the "Cancel" button.  This cancel button allows the abort option mentioned above.  This was added in the most recent version.

The second is that clicking anywhere within the message table while requests are being run will update the colored result output for whichever requests have finished thus far.

In the future a progress bar might be useful, but for now this is not a priority item.

**Comment by alias1** [comment]
Interesting. Perhaps I should pay more attention to the UI ;p
Updating the colours by click, is that a side effect of the Ui framework, or something you explicitly coded in? If explicitly, I wonder how hard it would be to update each request row after it's complete. Would have a similar effect to a progress bar (if not more useful), with potentially less changes required?

Get Outlook for iOS




On Tue, Nov 29, 2016 at 7:45 AM +1100, "mickayz" <notifications@github.com> wrote:












While it is less than ideal, the current version 0.6.2 has two UI indications for when requests are running.


The first is that the buttons on the bottom row will all be disabled aside from the "Cancel" button.  This cancel button allows the abort option mentioned above.  This was added in the most recent version.


The second is that clicking anywhere within the message table while requests are being run will update the colored result output for whichever requests have finished thus far.


In the future a progress bar might be useful, but for now this is not a priority item.

**Comment by mickayz** [comment]
This is something I had added custom.  Its currently done at line 1499.

At one point I had tried to do what you describe by inserting the following line of code at line 636: 

self.colorCodeResults()

While this accomplished the task of updating the results after each message run, it also had the undesirable side effect of a twitchy and unpleasant UI update. I decided to remove it for now, though there may be a way of tuning the table update to be more smooth.
